HW_paliiihor_12.py

```diff
@@ -0,0 +1,116 @@
+from functools import reduce
+
+"""Задача 1 10 баллов:
+Написать 3 примера генераторных функций с различными последовательностями."""
+
+
+def generator1():
+    print("HW 2,3")
+    yield "Monday"
+    print("HW 4,7")
+    yield "Tuesday"
+    print("HW  exercises 123,7")
+    yield "Wednesday"
+    print("HW Free")
+    yield "Thursday "
+    print("HW Free")
+    yield "Friday "
+    print("HW Free")
+    yield "Saturday"
+    print("HW Free")
+    yield "Sunday"
+
+
+g1 = generator1()
+try:
+    print(next(g1))
+    print(next(g1))
+    print(next(g1))
+    print(next(g1))
+    print(next(g1))
+    print(next(g1))
+    print(next(g1))
+    print(next(g1))
+except StopIteration as s:
+    print(s, "Week is finished")
+
+
+def generator2():
+    a = 0
+    while a != 10:
+        yield a + 1
+        a += 1
+
+
+g2 = generator2()
+print(next(g2))
+print(next(g2))
+print(next(g2))
+print(next(g2))
+
+g3 = (x for x in range(1, 10))
+print(next(g3))
+print(next(g3))
+print(next(g3))
+print(next(g3))
+
+"""Задача 2 10 баллов:
```

Comment:
https://docs.python.org/3.9/library/functools.html?highlight=reduce#functools.reduce 
Look at an example. Understand it and try to repeat it.